starter-core/src/test/groovy/io/micronaut/starter/application/ContextFactorySpec.groovy

```diff
@@ -0,0 +1,39 @@
+package io.micronaut.starter.application
```

Comment:
Currently, we default to Junit for Kotlin. Because of that, it passes without the order. 

Today, if we changed the default for Kotlin and set it to KoTest, Mockk is not automatically applied. 

I have added a test that verifies that test features are applied before features that depend on them are present.

I added a test: 

https://github.com/micronaut-projects/micronaut-starter/pull/1482/commits/0d3505acdde3c6a25affaab7a68743fcb723de08